test/extended/prometheus/image_pulls.go

```diff
@@ -0,0 +1,71 @@
+package prometheus
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	helper "github.com/openshift/origin/test/extended/util/prometheus"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/upgrades"
+)
+
+// ImagePullsAreFast asserts that image pulls complete quickly.
+type ImagePullsAreFast struct {
+	threshold time.Duration
+}
+
+func (t *ImagePullsAreFast) Name() string {
+	return "image-pulls-are-fast"
+}
+
+func (t *ImagePullsAreFast) DisplayName() string {
+	return "[sig-node] Image pulls are fast"
+}
+
+func (t *ImagePullsAreFast) Setup(_ *framework.Framework) {
+	if t.threshold.Seconds() == 0 {
+		t.threshold = 2 * time.Minute
+	}
+	return
+}
+
+func (t *ImagePullsAreFast) Test(f *framework.Framework, done <-chan struct{}, upgrade upgrades.UpgradeType) {
+	ctx := context.Background()
+	<-done
+
+	g.By(fmt.Sprintf("verifying that pull durations do not exceed %s", t.threshold))
+
+	oc := exutil.NewCLIWithFramework(f)
+
+	// we only consider samples since the beginning of the test
+	testDuration := exutil.DurationSinceStartInSeconds().String()
+
+	query := fmt.Sprintf(`
```

Comment:
This query could use a comment for explanation.

Reply:
added one with e9d6a9e0ab -> 688e4416da.